webapp/app/routes/app/proposals.$id/_route.tsx

```diff
@@ -163,6 +165,35 @@ export default function Proposals() {
       >
         <Await resolve={asyncData}>
           {({ addresses, proposal, reports, userSignedLegalVeto, userSignedProposal }) => {
+            const securityCouncilSignaturesReached =
+              proposal.signatures.approveUpgradeSecurityCouncil.length >=
+              NECESSARY_SECURITY_COUNCIL_SIGNATURES;
+            const guardiansSignaturesReached =
+              proposal.signatures.approveUpgradeGuardians.length >= NECESSARY_GUARDIAN_SIGNATURES;
+
+            const signProposalEnabled =
+              !userSignedProposal &&
+              proposal.status === PROPOSAL_STATES.Waiting &&
+              (user.role === "guardian" ? !proposal.guardiansApproval : true);
+            const signLegalVetoEnabled =
+              !userSignedLegalVeto &&
+              !proposal.guardiansExtendedLegalVeto &&
+              proposal.status === PROPOSAL_STATES.LegalVetoPeriod &&
+              user.role === "guardian";
+
+            const executeSecurityCouncilApprovalEnabled =
+              securityCouncilSignaturesReached && user.role === "securityCouncil";
+            const executeGuardiansApprovalEnabled =
+              guardiansSignaturesReached && user.role === "guardian";
+            const executeLegalVetoExtensionEnabled =
+              !proposal.extendedLegalVeto &&
+              proposal.status === PROPOSAL_STATES.LegalVetoPeriod &&
+              user.role === "guardian";
```

Comment:
this condition is not needed. Anyone can execute the transaction once the signatures are there.

Reply:
Execute legal veto can only be called by guardians: https://github.com/matter-labs/zk-governance/blob/654dd0c0b505e3140e7670470356f7edca83f19c/l1-contracts/src/ProtocolUpgradeHandler.sol#L234-L243